src/status_im/bots/events.cljs

```diff
@@ -52,25 +56,40 @@
 
 (def ^:private keywordize-vector (partial mapv keyword))
 
-;; TODO(janherich): do something with this horrible triple nested reduce so it's more readable
+(defn- transform-bot-subscriptions
+  "Transforms bot subscriptions as returned from jail in the following format:
+
+  `{:calculatedFee {:subscriptions {:value [\"sliderValue\"]
+                                    :tx [\"transaction\"]}}
+    :feeExplanation {:subscriptions {:value [\"sliderValue\"]}}}`
+
+  into data-structure better suited for subscription lookups based on changes
+  in `:bot-db`:
+
+  `{[:sliderValue] {:calculatedFee {:value [:sliderValue]
+                                    :tx [:transaction]}}
+    [:transaction] {:feeExplanation {:value [:sliderValue]}}}`
+
+  In the resulting data-structure, the top level keys are the (keywordized) paths
+  in the `:bot-db` data structure, so it's quick and easy to look-up all the
+  subscriptions which must be recalculated when something in their path changes."
+  [bot-subscriptions]
+  (reduce-kv (fn [acc sub-key {:keys [subscriptions]}]
+               (reduce-kv (fn [acc sub-param-key sub-param-path]
+                            (update acc (keywordize-vector sub-param-path)
+                                    assoc sub-key (utils/map-values subscriptions keywordize-vector)))
```

Comment:
We haven't discussed this, but for me it's usually hard to read lines longer than 80 chars. Can we extract at least inner `reduce`?

Reply:
Yes, I agree with you, it's still little bit hard to read, I will extract the inner `reduce-kv` to make it more readable.